client/src/selectors/profileSelectors.js

```diff
@@ -89,11 +84,32 @@ export const cadastralPageSelector: Selector<State, ContextRouter, number> = cre
   (page) => Number.parseInt(page, 10) || 1
 )
 
-export const paginatedCadastralInfoSelector = createSelector(
+export const cadastralSearchSelector: Selector<State, ContextRouter, string> = createSelector(
+  (_: State, props: ContextRouter): string =>
+    parseQueryFromLocation(props.location).cadastralSearch,
+  (search) => search || ''
+)
+
+export const filteredCadastralInfoSelector = createSelector(
   politicianCadastralSelector,
+  cadastralSearchSelector,
+  (cadastral, search) => {
+    return Object.values(cadastral).filter(({cadastralunitname}) =>
+      cadastralunitname.startsWith(search)
```

Comment:
you can call normalizeName on cadastralunitname, so that you don't have to add capitalization/punctuation in search